sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -536,12 +540,36 @@ private case class GpuParquetFileFilterHandler(@transient sqlConf: SQLConf) {
     convertToParquetNative(schema).asInstanceOf[ParquetFooter.StructElement]
   }
 
-  def readAndFilterFooter(
-      file: PartitionedFile,
-      conf : Configuration,
-      readDataSchema: StructType,
-      filePath: Path): ParquetFooter = {
-    val footerSchema = convertToFooterSchema(readDataSchema)
+  private def getFooterBuffer(
+      filePath: Path,
+      conf: Configuration,
+      metrics: Map[String, GpuMetric]): HostMemoryBuffer = {
+    val filePathString = filePath.toString
+    FileCache.get.getFooter(filePathString, conf).map { hmb =>
+      withResource(hmb) { _ =>
+        metrics.getOrElse(GpuMetric.FILECACHE_FOOTER_HITS, NoopMetric) += 1
+        metrics.getOrElse(GpuMetric.FILECACHE_FOOTER_HITS_SIZE, NoopMetric) += hmb.getLength
+        // buffer includes header and trailing length and magic, stripped here
+        hmb.slice(MAGIC.length, hmb.getLength - Integer.BYTES - MAGIC.length)
+      }
+    }.getOrElse {
+      metrics.getOrElse(GpuMetric.FILECACHE_FOOTER_MISSES, NoopMetric) += 1
+      withResource(readFooterBuffer(filePath, conf)) { hmb =>
+        metrics.getOrElse(GpuMetric.FILECACHE_FOOTER_MISSES_SIZE, NoopMetric) += hmb.getLength
+        // footer was not cached, so try to cache it
+        val cacheToken = FileCache.get.startFooterCache(filePathString)
+        cacheToken.foreach { t =>
+          t.complete(hmb.slice(0, hmb.getLength))
```

Comment:
I don't understand what the point of this is. You get back a cacheToken, and then need to call complete on it?  Some comments here would really be helpful. Later code makes it clearer what is happening, but still some comments would be helpful.